servicetalk-transport-netty-internal/src/main/java/io/servicetalk/transport/netty/internal/DefaultNettyConnection.java

```diff
@@ -470,10 +474,10 @@ private boolean failIfWriteActive(ChannelOutboundListener newChannelOutboundList
             // It is possible that we have set the writeSubscriber, then the channel becomes inactive, and we will
             // never notify the write writeSubscriber of the inactive event. So if the channel is inactive we notify
             // the writeSubscriber.
-            // It is also possible that Channel is in closing state, we should abort new writes if a closeReason was
-            // observed:
+            // It is also possible that Channel is in closing state, we should abort new writes from the client-side
+            // if a closeReason was observed:
             CloseEvent closeReason = null;
-            if (!channel().isActive() || (closeReason = this.closeReason) != null) {
+            if (!channel().isActive() || (isClient && (closeReason = this.closeReason) != null)) {
```

Comment:
Did you consider delaying the event from `CloseHandler` as that has most context of when it is correct to close. This propagates the close state machine understanding to the connection.

Reply:
Interesting idea, let me try

Reply:
There is a possibility to handle this scenario in `CloseHandler`, but it requires to remember the original close event and complicates the internal state machine, conflicting with #1141.
I would propose to go with this approach and rework it in a follow-up after #1141 is merged. Or work around this issue in #1141 by using TLSv1.2. It will simplify life a lot if we merge #1141 first, as it changes the state machine and introduces new important tests.